### ISSUE TITLE
[BUG] Update and Error events of old resources are skipped

### ISSUE THREAD

**Issue by aananthraj** [infracloudio/botkube#147]
**Describe the bug**
botkube skips the latest update, error event notifications of old resources available in the cluster. i.e, resources that were already existing before starting botkube.

**To Reproduce**
1. add config to watch update/error events in config.yaml
2. start/restart botkube server
3. change/modify the some old resources
4. notice that update/error events are skipped.

**Expected behavior**
Irrespective of the object creating time, botkube should send update, error events on resources when configured by the user.

**Snippet**
```bash
DEBU[2019-08-08T20:12:22+05:30] Processing update to service/app-5-demo-dummy in default namespaces 
DEBU[2019-08-08T20:12:22+05:30] event.TimeStamp >>> 0001-01-01 00:00:00 +0000 UTC 
DEBU[2019-08-08T20:12:22+05:30] Skipping older events
```
https://github.com/infracloudio/botkube/blob/d2d14cc026db4c3e5928a4c819650bc52199df33/pkg/events/events.go#L78-L87

`event.New()` initialises `event.time` with `0001-01-01 00:00:00 +0000 UTC` for events other than `CreateEvent` and `DeleteEvent` which is way long before `startTime` of botkube.

**note** : 
This is my first valuable bug. Feeling awesome :tada:
I wish I get some bug-bounty :rofl:

**Comment by PrasadG193** [comment]
Good catch @aananthraj  :). But I believe this is already fixed as a part of https://github.com/infracloudio/botkube/pull/146

**Comment by aananthraj** [comment]
> But I believe this is already fixed as a part of #146

Omg didnt see this comming through :eyes: 

Is this completely fixed.? 
I inspected for this bug and found it in #146 branch as well, before raising this issue and its corresponding fix.
can you check once by replicating the above mentioned steps..?

**Comment by PrasadG193** [comment]
@aananthraj I have tested this on #146 with mentioned steps and seems to be working as expected. Let's wait for #146 to get merged then try reproducing again.

**Comment by aananthraj** [comment]
sure, let me try now and update here.

**Comment by aananthraj** [comment]
@PrasadG193 , still the issue exists

```shell
DEBU[2019-08-09T15:53:53+05:30] Processing update to service/app-5-demo-dummy in default namespaces 
DEBU[2019-08-09T15:53:53+05:30] event.Timestamp >>> 0001-01-01 00:00:00 +0000 UTC 
DEBU[2019-08-09T15:53:53+05:30] Skipping older events                        
```
the following resources are still open to this issue

https://github.com/infracloudio/botkube/blob/d2d14cc026db4c3e5928a4c819650bc52199df33/pkg/events/events.go#L120-L123

https://github.com/infracloudio/botkube/blob/d2d14cc026db4c3e5928a4c819650bc52199df33/pkg/events/events.go#L140-L149

https://github.com/infracloudio/botkube/blob/d2d14cc026db4c3e5928a4c819650bc52199df33/pkg/events/events.go#L174-L181


let's fix this by initializing `event.Timestamp = time.Now()` to all events, which is the overwritten later based on the type of the event or kind of the object.

**Comment by PrasadG193** [comment]
Oh yes, you are right, let's fix this. Thanks a lot :slightly_smiling_face:

**Comment by aananthraj** [comment]
still, there are some issues with the event.New() implementation.

for instance, let's consider the following scenario 
- deployment object gets created and gets `obj.Status.Conditions` before botkube is started  
- botkube is started now
- an update event follows

https://github.com/infracloudio/botkube/blob/72f2f198421d7754062ab5604946bcecdf494999/pkg/events/events.go#L166-L172

**Problem:**
- since deployment object has `obj.Status.Conditions[condLen-1].LastTransitionTime.Time` as time before sarting of botkube server, recent `update event` will be skipped

we need to reopen this issue and fix all such issues.

**Comment by aananthraj** [comment]
> since deployment object has obj.Status.Conditions[condLen-1].LastTransitionTime.Time as time before sarting of botkube server, recent update event will be skipped

This issue will be fixed by #158
